Cython/Compiler/ExprNodes.py

```diff
@@ -12222,7 +12222,7 @@ def result_type(self, type1, type2, env):
             elif type1.is_pyunicode_ptr:
                 type1 = Builtin.unicode_type
             if type1.is_builtin_type or type2.is_builtin_type:
-                if type1 is type2 and self.operator in '**%+|&^':
+                if type1 is type2 and self.operator in '**%+&^':
                     # FIXME: at least these operators should be safe - others?
                     return type1
```

Comment:
Thinking about it some more, this "safe operators" assumption is always wrong for `type`. Let's just exclude it here.